static/panes/device-view.ts

```diff
@@ -0,0 +1,358 @@
+            labelField: 'name',
+            searchField: ['name'],
+            options: [],
+            items: [],
+            dropdownParent: 'body',
+            plugins: ['input_autogrow'],
+        });
+
+        if (state.irOutput) {
+            this.showDeviceAsmResults(state.irOutput);
+        }
+    }
+
+    override getInitialHTML(): string {
+        return $('#device').html();
+    }
+
+    override createEditor(editorRoot: HTMLElement): monaco.editor.IStandaloneCodeEditor {
+        return monaco.editor.create(
+            editorRoot,
+            monacoConfig.extendConfig({
+                language: 'asm',
+                readOnly: true,
+                glyphMargin: true,
+                lineNumbersMinChars: 3,
+            })
+        );
+    }
+
+    override registerOpeningAnalyticsEvent(): void {
+        ga.proxy('send', {
+            hitType: 'event',
+            eventCategory: 'OpenViewPane',
+            eventAction: 'DeviceAsm',
+        });
+    }
+
+    override registerEditorActions(): void {
+        this.editor.addAction({
+            id: 'viewsource',
+            label: 'Scroll to source',
+            keybindings: [monaco.KeyMod.CtrlCmd | monaco.KeyCode.F10],
+            keybindingContext: undefined,
+            contextMenuGroupId: 'navigation',
+            contextMenuOrder: 1.5,
+            run: ed => {
+                const position = ed.getPosition();
+                if (position != null) {
+                    const desiredLine = position.lineNumber - 1;
+                    const source = this.deviceCode[desiredLine].source;
+                    if (source && source.file == null && this.compilerInfo.editorId != null) {
+                        // a null file means it was the user's source
+                        this.eventHub.emit('editorLinkLine', this.compilerInfo.editorId, source.line, -1, -1, true);
+                    }
+                }
+            },
+        });
+    }
+
+    override registerCallbacks(): void {
+        this.linkedFadeTimeoutId = null;
+        const mouseMoveThrottledFunction = _.throttle(this.onMouseMove.bind(this), 50);
+        this.editor.onMouseMove(e => mouseMoveThrottledFunction(e));
+
+        const cursorSelectionThrottledFunction = _.throttle(this.onDidChangeCursorSelection.bind(this), 500);
+        this.editor.onDidChangeCursorSelection(e => cursorSelectionThrottledFunction(e));
+
+        this.selectize.on('change', this.onDeviceSelect.bind(this));
+
+        this.eventHub.on('colours', this.onColours, this);
+        this.eventHub.on('panesLinkLine', this.onPanesLinkLine, this);
+        this.eventHub.emit('deviceViewOpened', this.compilerInfo.compilerId);
+        this.eventHub.emit('requestSettings');
+
+        this.container.on('shown', this.resize, this);
+    }
+
+    override onCompileResult(id: number, compiler: CompilerInfo, result: CompilationResult): void {
+        if (this.compilerInfo.compilerId !== id) return;
+        // @ts-ignore
```

Comment:
Should instead use `@ts-expect-error` and it'd be awesome if you could add a tiny description in the comment about why it should fail and why it's fine